src/App.tsx

```diff
@@ -15,48 +16,121 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SORT_FEILD_ALPH = 'alph';
+const SORT_FEILD_LENGTH = 'length';
+
+function getFinalGoods(
+  goods: string[],
+  criterion: string,
+  direction: boolean,
+): string[] {
+  const preliminaryGoods = [...goods];
+
+  if (!criterion && direction) {
+    return preliminaryGoods.reverse();
+  }
+
+  if (criterion) {
+    preliminaryGoods.sort((good1, good2) => {
+      switch (criterion) {
+        case SORT_FEILD_LENGTH:
+          if (good1.length !== good2.length) {
+            return (direction)
+              ? good2.length - good1.length
+              : good1.length - good2.length;
+          }
+
+          return (direction)
+            ? good2.localeCompare(good1)
+            : good1.localeCompare(good2);
+
+        case SORT_FEILD_ALPH:
+          return (direction)
+            ? good2.localeCompare(good1)
+            : good1.localeCompare(good2);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return preliminaryGoods;
+}
```

Comment:
Your solution looks a little confusing, let's improve it
```suggestion
enum SortType {
  NONE,
  ALPHABET,
  LENGTH,
}
type ReorderOptions = {
  sortType: SortType,
  isReversed: boolean,
};

export function getReorderedGoods(
  goods: string[],
  { sortType, isReversed }: ReorderOptions,
) {
  const visibleGoods = [...goods];

  visibleGoods.sort((good1, good2) => {
    switch (sortType) {
      case (SortType.ALPHABET):
        return good1.localeCompare(good2);

      case (SortType.LENGTH):
        return good1.length - good2.length;

      default:
        return 0;
    }
  });

  if (isReversed) {
    visibleGoods.reverse();
  }

  return visibleGoods;
}
```